gnd/src/main/res/layout/location_lock_btn.xml

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!--
+  ~ Copyright 2018 Google LLC
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     https://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto">
+
+  <data>
+    <variable
+      name="viewModel"
+      type="com.google.android.gnd.ui.home.LocationLockViewModel" />
+  </data>
+
+  <com.google.android.material.floatingactionbutton.FloatingActionButton
+    android:id="@+id/location_lock_btn"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:enabled="@{viewModel.locationLockEnabled}"
```

Comment:
Huh, I don't think I've seen this syntax in XML before, is that Kotlin?

Reply:
This is data binding. (not related to Kotlin)

https://developer.android.com/topic/libraries/data-binding

Reply:
Gotcha, thanks for the info!